### PR TITLE
GraphID

### PR DESCRIPTION
**Pull request by lwsanty**
related to [[DEV-3075]]

[DEV-3075]: https://athenianco.atlassian.net/browse/DEV-3075?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: github/types/event.go [hunk 1]
```diff
@@ -19,9 +19,10 @@ const (
 )
 
 type RenameEvent struct {
-	NodeID  string `json:"node_id"`
-	Name    string `json:"name"`
-	NameOld string `json:"name_old,omitempty"`
+	NodeID  string  `json:"node_id"`
```
Comment (dennwc) [inline github/types/event.go:22]: Should also become `omitempty` now.

File: github/types/ghtypes.go [hunk 2]
```diff
@@ -68,6 +70,77 @@ type EventContext struct {
...
+	return NewGraphID(id, typ), nil
+}
+
+// GraphID is an integer graph node ID used in Athenian.
+type GraphID struct {
+	id  uint64
+	typ string
+}
+
+func (GraphID) isValue() {}
```
Comment (dennwc) [inline github/types/ghtypes.go:105]: Please remove, it does nothing.

File: github/types/event.go [hunk 3]
```diff
@@ -32,6 +33,7 @@ type RepoEvent struct {
 	OrgName      string        `json:"org_name,omitempty"`   // set for OrgRenamed; deprecated
 	OrgRename    *RenameEvent  `json:"org_rename,omitempty"` // set for OrgRenamed events
 	NodeIDs      []NodeID      `json:"node_id,omitempty"`
+	GIDs         []GraphID     `json:"gid,omitempty"`
```
Comment (dennwc) [inline github/types/event.go:36]:
```suggestion
	GIDs         []GraphID     `json:"gids,omitempty"`
```

Comment (lwsanty) [reply github/types/event.go]: Done.

**Pull request merged**